transformer_engine/common/cast/mxfp8/specialized/gated_mxfp8_rowwise_swiglu.cuh

```diff
@@ -0,0 +1,779 @@
+    asm volatile(
+        "{\n\t"
+        ".reg.b64 in_act, in_gate, in_grad, s; \n\t"
+        "mov.b64 in_act, %6; \n\t"
+        "mov.b64 in_gate, %7; \n\t"
+        "mov.b64 in_grad, %8; \n\t"
+        "mov.b64 s, %9; \n\t"
+
+        ".reg.b64 act; \n\t"
+        "mul.f32x2 act, in_act, s; \n\t"  // x * s
+        ".reg.f32 one; \n\t"
+        "mov.f32 one, 1.0; \n\t"
+        ".reg.b64 ones; \n\t"
+        "mov.b64 ones, {one, one}; \n\t"
+        ".reg.b64 sub1s; \n\t"
+        "sub.f32x2 sub1s, ones, s; \n\t"  // (1 - s)
+        ".reg.b64 dact; \n\t"
+        "fma.rn.f32x2 dact, act, sub1s, s; \n\t"  // act * (1 - s) + s
+        ".reg.b64 out_act; \n\t"
+        ".reg.b64 out_gate; \n\t"
+        ".reg.b64 grad_gate; \n\t"
+        "mul.f32x2 grad_gate, in_gate, in_grad; \n\t"
+        "mul.f32x2 out_act, dact, grad_gate; \n\t"
+        "mul.f32x2 out_gate, act, in_grad; \n\t"
+        ".reg.f32 out_act1, out_act2; \n\t"
+        ".reg.f32 out_gate1, out_gate2; \n\t"
+        "mov.b64 {out_act1, out_act2}, out_act; \n\t"
+        "mov.b64 {out_gate1, out_gate2}, out_gate; \n\t"
+        ".reg.b32 cvt_out_act, cvt_out_gate; \n\t"
+        "cvt.rn.satfinite.f16x2.f32 cvt_out_act, out_act2, out_act1; \n\t"
+        "cvt.rn.satfinite.f16x2.f32 cvt_out_gate, out_gate2, out_gate1; \n\t"
+        "mov.b32 %0, cvt_out_act; \n\t"
+        "mov.b32 %1, cvt_out_gate; \n\t"
+        "max.xorsign.abs.f16x2 %2, %4, cvt_out_act; \n\t"
+        "max.xorsign.abs.f16x2 %3, %5, cvt_out_gate; \n\t"
+        "}\n"
+        : "=r"(reinterpret_cast<uint32_t &>(out_act)), "=r"(reinterpret_cast<uint32_t &>(out_gate)),
+          "=r"(reinterpret_cast<uint32_t &>(out_xormax_act)),
+          "=r"(reinterpret_cast<uint32_t &>(out_xormax_gate))
+        : "r"(reinterpret_cast<const uint32_t &>(in_xormax_act)),
+          "r"(reinterpret_cast<const uint32_t &>(in_xormax_gate)),
+          "l"(reinterpret_cast<const uint64_t &>(in_act)),
+          "l"(reinterpret_cast<const uint64_t &>(in_gate)),
+          "l"(reinterpret_cast<const uint64_t &>(in_grad)),
+          "l"(reinterpret_cast<const uint64_t &>(s)));
+  }
+#else
+  NVTE_DEVICE_ERROR("compute_bwd_gated_swiglu is only supported on SM 10.0+.");
+#endif
+}
+
+template <typename IType2>
+__device__ __forceinline__ void compute_bwd_gated_swiglu(IType2 &out_act, IType2 &out_gate,
+                                                         IType2 &out_xormax_act,
+                                                         IType2 &out_xormax_gate,
+                                                         const float2 in_act, const float2 in_gate,
+                                                         const float2 in_grad) {
+#if (defined __CUDA_ARCH__) && (__CUDA_ARCH__ >= 1000)
+  const IType2 in_xormax_act = out_xormax_act;
+  const IType2 in_xormax_gate = out_xormax_gate;
+
+  const float2 s{sigmoidf(in_act.x), sigmoidf(in_act.y)};
+
+  constexpr bool USE_TUNED_SWIGLU_PTX = true;
```

Comment:
Why would we ever want to not set it to true? If this is just from the development please remove it and leave only the tuned path.